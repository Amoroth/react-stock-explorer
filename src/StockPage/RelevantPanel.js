import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

import styles from './StockPage.module.css'

const relevantPanel = ({ name, change, symbol, close, onSelect }) => {
  let shortCompanyName = name
  const priceChange = `${change > 0 ? '+' : ''}${(
    change * 100
  ).toFixed(2)}%`

  if (shortCompanyName.length > 14) {
    shortCompanyName = `${shortCompanyName.slice(0, 12)}...`
  }

  return (
    <div className={styles['relavent-item']} key={symbol}>
      <div
        className={`${styles['relavent-item-row']} ${
          styles['relavent-item-title']
        }`}
      >
        <h6
          title={
            shortCompanyName[shortCompanyName.length - 2] === '.'
              ? name
              : null
          }
        >
          {shortCompanyName}
        </h6>
        <span>{symbol}</span>
      </div>
      <div
        className={`${styles['relavent-item-row']} ${
          styles['relavent-item-info']
        }`}
      >
        <span>{close}</span>
        <span
          style={{ color: change > 0 ? 'green' : 'red' }}
        >
          {priceChange}
        </span>
        <Link
          to={`/stock?cmp=${symbol}`}
          onDragStart={(e) => e.preventDefault()}
          onClick={() => onSelect(symbol)}
        >
          More...
        </Link>
      </div>
    </div>
  )
}

relevantPanel.propTypes = {
  name: PropTypes.string.isRequired,
  change: PropTypes.number.isRequired,
  symbol: PropTypes.string.isRequired,
  close: PropTypes.number.isRequired,
  onSelect: PropTypes.func.isRequired,
}

export default relevantPanel
