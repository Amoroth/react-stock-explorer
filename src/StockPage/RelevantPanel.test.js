import React from 'react'
import { shallow } from 'enzyme'

import RelevantPanel from './RelevantPanel'

describe('<RelevantPanel />', () => {
  let props
  let wrapper

  beforeEach(() => {
    props = {
      name: 'Company',
      change: 0.04,
      symbol: 'COM',
      close: '4 USD',
      onSelect: jest.fn(),
    }
    wrapper = shallow(<RelevantPanel {...props} />)
  })

  it('renders', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('sends to new company from link', () => {
    wrapper.find('Link').simulate('click')
    
    expect(props.onSelect).toBeCalled()
  })

  it('formats and styles percentage', () => {
    expect(wrapper.find('.relavent-item-info span').last().text()).toBe('+4.00%')
    expect(wrapper.find('.relavent-item-info span').last().prop('style')).toEqual({ color: 'green' })

    wrapper.setProps({ change: -0.06 })

    expect(wrapper.find('.relavent-item-info span').last().text()).toBe('-6.00%')
    expect(wrapper.find('.relavent-item-info span').last().prop('style')).toEqual({ color: 'red' })
  })

  it('shortens title if too long', () => {
    wrapper.setProps({ name: 'Company with really long name' })

    expect(wrapper.find('h6').text()).toBe('Company with...')
  })

  it('does not set title if title is short', () => {
    expect(wrapper.find('h6').prop('title')).toBe(null)
  })

  it('sets title if title is too long', () => {
    const longName = 'Company with really long name'
    wrapper.setProps({ name: longName })

    expect(wrapper.find('h6').prop('title')).toBe(longName)
  })
})
