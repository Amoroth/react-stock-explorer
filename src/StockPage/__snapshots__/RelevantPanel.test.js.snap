// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RelevantPanel /> renders 1`] = `
<div
  className="relavent-item"
  key="COM"
>
  <div
    className="relavent-item-row relavent-item-title"
  >
    <h6
      title={null}
    >
      Company
    </h6>
    <span>
      COM
    </span>
  </div>
  <div
    className="relavent-item-row relavent-item-info"
  >
    <span>
      4 USD
    </span>
    <span
      style={
        Object {
          "color": "green",
        }
      }
    >
      +4.00%
    </span>
    <Link
      onClick={[Function]}
      onDragStart={[Function]}
      replace={false}
      to="/stock?cmp=COM"
    >
      More...
    </Link>
  </div>
</div>
`;
